### PR TITLE
refactor(core): Decouple post workflow execute event from internal hooks (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Follow-up to: #10221

### CODE REVIEW

File: packages/cli/src/WorkflowExecuteAdditionalData.ts [hunk 1]
```diff
@@ -646,13 +646,18 @@ function hookFunctionsSaveWorker(): IWorkflowExecuteHooks {
...
 				const { executionId, workflowData: workflow } = this;
 
-				void internalHooks.onWorkflowPostExecute(executionId, workflow, runData);
 				eventService.emit('workflow-post-execute', {
-					workflowId: workflow.id,
-					workflowName: workflow.name,
-					executionId,
-					success: runData.status === 'success',
-					isManual: runData.mode === 'manual',
+					audit: {
```
Comment (netroy) [inline packages/cli/src/WorkflowExecuteAdditionalData.ts:650]: why not emit two separate events?

File: packages/cli/src/WorkflowExecuteAdditionalData.ts [hunk 2]
```diff
@@ -934,14 +939,20 @@ async function executeWorkflow(
...
 
-	void internalHooks.onWorkflowPostExecute(executionId, workflowData, data, additionalData.userId);
 	eventService.emit('workflow-post-execute', {
-		workflowId: workflowData.id,
-		workflowName: workflowData.name,
-		executionId,
-		success: data.status === 'success',
-		isManual: data.mode === 'manual',
-		userId: additionalData.userId,
+		audit: {
```
Comment (netroy) [inline packages/cli/src/WorkflowExecuteAdditionalData.ts:943]: same here.

**Review by netroy — CHANGES_REQUESTED** [review]
perhaps we should prefix all telemetry events with `telemetry.` and audit events with `audit.` instead of trying to use the same event for both.

File: packages/cli/src/WorkflowExecuteAdditionalData.ts (hunk 1 above)
Comment (tomi) [reply packages/cli/src/WorkflowExecuteAdditionalData.ts:650]: Other option would be to keep the event payload generic enough so both `telemetry-event-relay` and `audit-event-relay` can then derive whatever data they need from that. Tho that makes it then implicit what happens based on the sent event. Then again that's usually the point of using event buses: to decouple consumers from producers. Now the coupling is there, as we explicitly define the data separately for each consumer.

**Comment by ivov** [comment]
> prefix all telemetry events

I thought of prefixing event names but ruled this out, because there is only a small minority of events that are sent to both relays with different payloads, because this adds coupling to all events and relays, and because I expect that sending multiple events with different payloads at all callsites will be noisy.

For these edge cases I'd like to aim for unifying both payloads as they're usually similar enough, but that would require a breaking change. Sending a big payload from which relays pick what they need is error prone (we have no tests) and causes the payload type to be unclear. Payloads with nesting in edge cases like this (although adding a bit of coupling) are meant as a workaround until we can unify.

File: packages/cli/src/eventbus/event.types.ts [hunk 3]
```diff
@@ -310,4 +300,21 @@ export type Event = {
...
+		executionId: string;
+		success: boolean;
+		workflowId: string;
+		isManual: boolean;
+		workflowName: string;
+		metadata?: Record<string, string>;
+
+		// telemetry only
+		workflow: IWorkflowBase;
+		runData?: IRun;
```
Comment (tomi) [inline packages/cli/src/eventbus/event.types.ts:318]: This could be simplified to

```suggestion
		userId?: string;
		executionId: string;
		workflow: IWorkflowBase;
		runData?: IRun;
```

as `metadata` doesn't seem to be passed in any `emit` and all the other fields can be derived from `workflow` and `runData`. That way there's no need to duplicate the logic for deriving those fields

**Comment by ivov** [comment]
Too many conflicts to safely resolve, I'll reopen from master soon enough.
